iot-hub/Samples/device/MessageReceiveSample/Parameters.cs

```diff
@@ -21,5 +21,12 @@ internal class Parameters
             Required = false,
             HelpText = "The transport to use to communicate with the IoT Hub. Possible values include Mqtt, Mqtt_WebSocket_Only, Mqtt_Tcp_Only, Amqp, Amqp_WebSocket_Only, Amqp_Tcp_only, and Http1.")]
         public TransportType TransportType { get; set; }
+
+        [Option(
+            'r',
```

Comment:
Now this sample supports the `-r` param, just like the reconnection sample. The default is null, which equates to run until explicitly canceled by the user using control+c.